drivers/hmis/app/models/hmis/unit.rb

```diff
@@ -162,6 +162,26 @@ def to_pick_list_option
     { code: id, label: display_name }
   end
 
+  # Build an unsaved CE Opportunity for this unit
+  def build_ce_opportunity(rule_resolver: Hmis::Ce::Match::UnitGroupRuleResolver.new)
+    raise 'Unit already has an active opportunity' if latest_opportunity&.active?
+    raise 'Unit must be in a Unit Group to build CE opportunity' unless unit_group
+    raise 'Unit Group has no Workflow Template' unless unit_group.workflow_template || unit_group.direct_referral_workflow_template
+
+    unit_desc = unit_type&.description
+    opportunity_name = "Unit #{id}#{unit_desc ? ' - ' : ''}#{unit_desc}"
+
+    rules = rule_resolver.rules_for_unit_group(unit_group)
+
+    Hmis::Ce::Opportunity.new(
+      unit: self,
```

Comment:
Since the name of this method mimics the built-in `build..` helpers, I think it would make sense to build this onto the unit directly since that is the typical expectation of `build..` methods:
```suggestion
    opportunities.build(
```